plasma_framework/docs/integration-docs/integration-doc.md

```diff
@@ -1036,4 +1036,140 @@ Exit games allow users to participate in ensuring the correctness of the system.
+```
+event VaultRegistered(
+    uint256 vaultId, // id of the vault in the Plasma Framework
+    address vaultAddress // address of the vault
+);
+```
+
+### Event SetDepositVerifierCalled (emitted in vault contracts)
+
+Chain maintainer chan change deposit verification for a vault by switching to a new deposit verifier, which is a contract that verifies validity of a Plasma deposit. Change of a verifier takes effect after 2 exit periods. User can inspect the source code of the new deposit verifier to examine whether it is safe to use. 
+```
+event VaultRegistered(
+    uint256 vaultId, // id of the vault in the Plasma Framework
+    address vaultAddress // address of the vault
+);
+```
+
+### Event DepositCreated (emitted in EthVault contract)
+
+After a successful deposit of ETH, `EthVault` contract emits `DepositCreated` event. This event notifies that deposited funds are available to spend.
+```
+event DepositCreated(
+    address indexed depositor, // address that deposited the funds
+    uint256 indexed blknum, // Plasma block number that contains the deposit
+    address indexed token, // ETH
+    uint256 amount // amount of ETH deposited
+);
+```
+
+### Event DepositCreated (emitted in Erc20Vault contract)
+
+After a successful deposit of ERC20 tokens, `Erc20Vault` contract emits `DepositCreated` event. This event notifies that deposited funds are available to spend.
+```
+event DepositCreated(
+    address indexed depositor, // address that deposited the funds
+    uint256 indexed blknum, // Plasma block number that contains the deposit
+    address indexed token, // deposited ERC20 token 
+    uint256 amount // amount of tokens deposited
+);
+```
+
+### Event EthWithdrawn (emitted in EthVault contract)
+
+After a successful withdawal of ETH, `EthVault` contract emits `EthWithdrawn` event. This event notifies that withdrawn funds are no longer available to spend on Plasma chain and have been transfered to the receiver.
+```
+event EthWithdrawn(
+    address indexed receiver, // address that receives the funds
+    uint256 amount // amount withdrawn
+);
+```
+
+### Event Erc20Withdrawn (emitted in Erc20Vault contract)
+
+After a successful withdawal of ERC20 tokens, `Erc20Vault` contract emits `Erc20Withdrawn` event. This event notifies that withdrawn funds are no longer available to spend on Plasma chain and have been transfered to the receiver.
+```
+event Erc20Withdrawn(
+    address indexed receiver, // address that receives the funds
+    address indexed token, // transferred ERC20 token
+    uint256 amount // amount withdrawn
+);
+```
+
+### Event WithdrawFailed (emitted in EthVault contract)
+
+If ETH withdrawal failed `EthVault` contract emits `WithdrawFailed` event. This event notifies that withdrawal, which is a result of an exit, failed. That event is a result of malicious user or vault not having enough funds, most probably result of an malicious attack.
```

Comment:
```suggestion
If a withdrawal fails, the `EthVault` contract emits a `WithdrawFailed` event. This event notifies that a withdrawal, which is a result of an exit, failed. This can happen because of malicious behaviour or the vault not having enough funds, most probably due to a malicious attack.
```